tests/std/tests/P0896R4_istream_view/test.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <iostream>
+#include <ranges>
+#include <sstream>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+constexpr int expected_empty[] = {-1, -1, -1, -1, -1};
+constexpr int expected[]       = {0, 1, 2, 3, -1};
+
+struct streamable {
+    streamable() = default;
+    streamable(const int input) : _val(input) {}
+
+    friend istream& operator>>(istream& is, streamable& right) noexcept {
+        is >> right._val;
+        return is;
+    }
+
+    friend bool operator==(const streamable& left, const streamable& right) noexcept = default;
+
+    int _val = 0;
+};
+
+template <class T>
+void test_one_type() {
+    using ranges::basic_istream_view;
+
+    // validate type properties
+    using R = basic_istream_view<T, char>;

```

Comment:
I observe that this doesn't test character types other than `char`, much less custom traits, so there are certain kinds of product bugs that this could miss - however, the product code appears to be correct, so I don't think such test coverage is necessary to proceed. Something to keep in mind for the future, though.

Reply:
Yes, the thought was that every wild character type would actually only implement the `basic_istream` functionality not the `istream_view` itself

Reply:
I'm mostly concerned about typos where we should have said `_Elem` but instead said `char`, so it doesn't work with `wchar_t`.